packages/backend/src/graphql/projectMembership/mutations/projectMembershipCreateByEmailMutationField.ts

```diff
@@ -0,0 +1,46 @@
+import { builder } from '../../builder'
+import { prisma } from '../../prisma'
+
+builder.mutationField('projectMembershipCreateByEmail', (t) =>
+  t.prismaField({
+    type: 'Project',
+    description: 'Assign user to a project by e-mail.',
+    args: {
+      email: t.arg.string(),
+      projectId: t.arg.id(),
+    },
+    authScopes: (_source, { projectId }) => ({ isAdminByProject: projectId.toString() }),
+    resolve: async (query, _source, { email, projectId }) => {
+      const user = await prisma.user.findUnique({
+        where: { email },
+      })
+      if (!user) {
+        throw new Error(`No user found with email ${email}`)
+      }
+      const userId = user.id
+
+      const existingMembership = await prisma.projectMembership.findFirst({
+        where: {
+          userId: userId,
+          projectId: projectId.toString(),
+        },
+      })
+
+      if (existingMembership) {
+        throw new Error(`User is already a member of the project`)
+      }
+
+      const projectMembership = await prisma.projectMembership.upsert({
```

Comment:
You can now just use `.create()`, as you already handled the case when a project membership already exists.